iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/BranchData.java

```diff
@@ -159,6 +155,18 @@ public BranchData(Line line, double epsilonX, boolean applyReactanceCorrection)
         mainComponent1 = bus1 != null ? bus1.isInMainConnectedComponent() : connectableMainComponent1;
         mainComponent2 = bus2 != null ? bus2.isInMainConnectedComponent() : connectableMainComponent2;
 
+        if (isStructuralRatioLineNeeded(line)) {
+            rho1 = 1.0 / structuralRatioEnd1(line);
+            rho2 = 1f;
```

Comment:
Some duplicated code here, a small factorization could be done.

Reply:
Done